synapse/storage/database.py

```diff
@@ -584,7 +642,10 @@ def inner_func(conn, *args, **kwargs):
                     logger.debug("Reconnecting closed database connection")
                     conn.reconnect()
 
-                return func(conn, *args, **kwargs)
+                db_conn = LoggingDatabaseConnection(
+                    conn, self.engine, "runWithConnection"
+                )
+                return func(db_conn, *args, **kwargs)
```

Comment:
Is this ensuring another location uses `LoggingDatabaseConnection` or is it necessary because of the refactoring? I'm having a bit of trouble following why this change was done.

Reply:
This just ensures that whenever we have use a connection after a call to `runWithConnection` it is a logging connection, which a) helps ensure that more SQL statements get logged and b) necessary to make the tests work  as they call `runWithConnection` and then call things expecting a logging connection a lot.